util/testutil/integration/dockerd.go

```diff
@@ -197,21 +223,112 @@ func waitForAPI(ctx context.Context, apiClient *client.Client, d time.Duration)
 	return nil
 }
 
+const (
+	DockerdSkipCacheExport      = "cache export"
+	DockerdSkipCacheImport      = "cache import"
+	DockerdSkipDirectPush       = "direct push"
+	DockerdSkipFrontendOutline  = "frontend.outline"
+	DockerdSkipFrontendTargets  = "frontend.targets"
+	DockerdSkipImageExporter    = "image exporter"
+	DockerdSkipInfo             = "info"
+	DockerdSkipMultiCacheExport = "multi cache export"
+	DockerdSkipMultiPlatform    = "multi-platform"
+	DockerdSkipOCIExporter      = "oci exporter"
+	DockerdSkipSecurityMode     = "security mode"
+	DockerdSkipSourceDateEpoch  = "source date epoch"
+)
+
+var (
+	dockerdMobyFeaturesEnabled = map[string]bool{
+		DockerdSkipCacheExport:      false,
+		DockerdSkipCacheImport:      false,
+		DockerdSkipDirectPush:       false,
+		DockerdSkipFrontendOutline:  false,
+		DockerdSkipFrontendTargets:  false,
+		DockerdSkipImageExporter:    false,
+		DockerdSkipInfo:             false,
+		DockerdSkipMultiCacheExport: false,
+		DockerdSkipMultiPlatform:    false,
+		DockerdSkipOCIExporter:      false,
+		DockerdSkipSecurityMode:     false,
+		DockerdSkipSourceDateEpoch:  false,
+	}
+	dockerdSnapshotterFeaturesEnabled = map[string]bool{
+		DockerdSkipCacheExport:      true,
+		DockerdSkipCacheImport:      true,
+		DockerdSkipDirectPush:       true,
+		DockerdSkipFrontendOutline:  false,
+		DockerdSkipFrontendTargets:  false,
+		DockerdSkipImageExporter:    true,
+		DockerdSkipInfo:             false,
+		DockerdSkipMultiCacheExport: true,
+		DockerdSkipMultiPlatform:    true,
+		DockerdSkipOCIExporter:      true,
+		DockerdSkipSecurityMode:     false,
+		DockerdSkipSourceDateEpoch:  false,
+	}
+)
+
+func isDockerdTestFeatureEnabled(feature string, snapshotter bool) *bool {
+	features := dockerdMobyFeaturesEnabled
+	if snapshotter {
+		features = dockerdSnapshotterFeaturesEnabled
+	}
+	if enabled, ok := features[feature]; ok {
+		return &enabled
+	}
+	return nil
+}
+
 func SkipIfDockerd(t *testing.T, sb Sandbox, reason ...string) {
 	t.Helper()
-	sbx, ok := sb.(*sandbox)
-	if !ok {
-		t.Fatalf("invalid sandbox type %T", sb)
+	b, err := getBackend(sb)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !b.isDockerd {
+		return
+	}
+	if len(reason) == 0 {
+		t.Fatal("no reason provided")
 	}
-	b, ok := sbx.Backend.(backend)
-	if !ok {
-		t.Fatalf("invalid backend type %T", b)
+
+	snapsh := "moby"
+	if b.snapshotter != "" {
+		snapsh = b.snapshotter
+	}
+
+	var ereasons []string
+	for _, r := range reason {
```

Comment:
there should be a check in here that reason always needs to be in the defined set.